pkg/network/setup/netpod/discover.go

```diff
@@ -63,6 +63,8 @@ func (n NetPod) discover(currentStatus *nmstate.Status) error {
 				return err
 			}
 
+			// This cache is no longer used by vit-launcher, the dummy interface is used instead to store the data.
```

Comment:
We leave it, the cache wasn't removed. It is just not used by the virt-launcher.

Reply:
I mean,  the file remain so use it for what we can as of toady, instead of passing data via the dummy interface.

In general I am against using the dummy interface for passing data to from virt-handler to virt-launcher.
It feel wrong and as a reviewer its not intuitive and seem like a hack.
Also, it may break if kubelet behavior change in regard to pod networking readiness.

I prefer if we could use the same method as today, pass data via file (cache), and discuss this change and alternatives (e.g.: pass data through, downward API, handler-launcher-com gRPC channel).

Its not clear to me what will be the implication of this change (and others that will use the dummy iface for passing data) in terms of backward compatibility.
Do you think it will be possible to get rid of of the dummy interface eventually?

Reply:
Then bridge binding won't be able to use the tap domain attachment (which knows nothing about cache).
I don't understand why cache file is better than a dummy interface. We are trying to move the phase1 to CNIs. CNIs have access only to the pod network space, so using a dummy interface to store data fits very well to this approach.

> Do you think it will be possible to get rid of of the dummy interface eventually?

As long as kubelet needs it, no. For passing data between the phases, maybe in the future we will find another solution. But I don't see it as a temporary solution.

Reply:
>  I don't understand why cache file is better than a dummy interface. We are trying to move the phase1 to CNIs. CNIs have access only to the pod network space, so using a dummy interface to store data fits very well to this approach.

Passing data via file is well known and straight forward then passing data via dummy interface.
Its better to use files because it is expected, the next reviewer/dev would understand how the subject info passed down because its a common pattern.

> We are trying to move the phase1 to CNIs. CNIs have access only to the pod network space, so using a dummy interface to store data fits very well to this approach.

I recall the bridge binding is planed to stay part of Kubevirt core, isn't that right? 
If so it could access to files in the pod, similar to dummy interfaces.
If the bridge binding is converted to self-contained CNI, wouldn't it get executed before the dummy interface is created (by phase1 code)?

Reply:
> Passing data via file is well known and straight forward then passing data via dummy interface. Its better to use files because it is expected, the next reviewer/dev would understand how the subject info passed down because its a common pattern.

I am not familiar with files being a common pattern to pass information between processes.
I would say it is a bad pattern, full of assumptions and hacks with hard to manage comparability for changes.
But all this is not the main reason we want to get rid of it.
 
> > We are trying to move the phase1 to CNIs. CNIs have access only to the pod network space, so using a dummy interface to store data fits very well to this approach.
> 
> I recall the bridge binding is planed to stay part of Kubevirt core, isn't that right? If so it could access to files in the pod, similar to dummy interfaces. 

We are building a binding plugin infrastructure, so even if one specific binding is planned to remain in kubevirt core, we still need to allow community members to build similar bindings (with special tweaks) successfully.

> If the bridge binding is converted to self-contained CNI, wouldn't it get executed before the dummy interface is created (by phase1 code)?

Like with the coming `passt` bindings, the kubevirt CNI addition is executed after the main interface CNI is executed (e.g. `eth0`). The dummy is created as part of the "config" stage.
Then the information persisted on the dummy interface can be used in the virt-launcher step, by reading it. In practice, the author of the bridge binding CNI would be the same author of the bridge binding sidecar.
As we cannot share files between the two stages, the dummy seems a reasonable solution.
As we keep the old files for backward compatability, it makes this change revertible. If we find a problem, we could revert to use the files again.

Do you see a case where this method will fail to serve this data communication need?

Reply:
> I am not familiar with files being a common pattern to pass information between processes.
I would say it is a bad pattern, full of assumptions and hacks with hard to manage comparability for changes.
But all this is not the main reason we want to get rid of it.

The same can be claimed on using dummy interface for storing data, at least it is expected from files to..store date.

> We are building a binding plugin infrastructure, so even if one specific binding is planned to remain in kubevirt core, we still need to allow community members to build similar bindings (with special tweaks) successfully.

I talked about the example Alone raised, which is the core bridge binding.
Also, the binding plugin CNI author need to know about passing info to Kubevirt using dummy interface.

> As we cannot share files between the two stages, the dummy seems a reasonable solution.

The binding CNIs could save the original interface info in a file on the node, or add annotation to the pod with relevant info.
virt-handler will then to pass the info using file virt-launcher would then read (similar to the DHCP config cache).
It require some more thinking and testing but I think its feasible.

Reply:
> > I am not familiar with files being a common pattern to pass information between processes.
> > I would say it is a bad pattern, full of assumptions and hacks with hard to manage comparability for changes.
> > But all this is not the main reason we want to get rid of it.
> 
> The same can be claimed on using dummy interface for storing data, at least it is expected from files to..store date.

Indeed, that is why it is not the main reason.

> I talked about the example Alone raised, which is the core bridge binding. Also, the binding plugin CNI author need to know about passing info to Kubevirt using dummy interface.

The author needs to know about the network binding plugin API, how it integrates with kubevirt, etc.
The network binding plugin original goal is to provide means to implement something like the bridge binding, therefore the core bindings are less relevant in terms of how they work. We need something that works for everything, so we will have one solution that fits all.
Obviously if there needs to be an exception, we could have it.

> > As we cannot share files between the two stages, the dummy seems a reasonable solution.
> 
> The binding CNIs could save the original interface info in a file on the node, or add annotation to the pod with relevant info. virt-handler will then to pass the info using file virt-launcher would then read (similar to the DHCP config cache). It require some more thinking and testing but I think its feasible.

It is a possible solution indeed. However, due to its complexity, I would not choose it as my first candidate if there are other simpler ones. Using annotations can be in practice done without kubevirt intervention, so if some author decides it fits his needs better, he could do it like that.
Anyway, these kind of options better be raised in the design proposal, we should record all ideas raised anyway.